compiler/packages/babel-plugin-react-compiler/src/Entrypoint/Program.ts

```diff
@@ -430,11 +445,52 @@ export function compileProgram(
         prunedMemoValues: compiledFn.prunedMemoValues,
       });
     } catch (err) {
+      if (body.isBlockStatement()) {
+        const optOutDirectives = findDirectiveDisablingMemoization(body);
+        if (optOutDirectives.length > 0) {
+          logError(err, pass, fn.node.loc ?? null);
+          return null;
+        }
+      }
```

Comment:
If we compile a function and find that there is at least one opt out directive in its body, we log the error instead of throwing. This preserves the opt out behavior (it doesn't get compiled) while still allowing the eslint plugin to surface any errors that may have been lurking underneath